packages/zudoku/scripts/generate-icon-types.ts

```diff
@@ -1,9 +1,12 @@
 import { readFileSync, writeFileSync } from "node:fs";
+import { createRequire } from "node:module";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
 import { format } from "prettier";
 import * as ts from "typescript";
 
-const path = import.meta.resolve("lucide-react/dynamicIconImports.js");
-const filePath = new URL(path).pathname;
+const require = createRequire(import.meta.url);
+const filePath = require.resolve("lucide-react/dynamicIconImports.js");
```

Comment:
Why is this change necessary? What's the difference between `import.meta.resolve` and `require.resolve`?

Reply:
> Why is this change necessary? What's the difference between `import.meta.resolve` and `require.resolve`?

The change from using `import.meta.resolve` to `require.resolve` is necessary because `import.meta.resolve` is not a standard feature in Node.js, whereas `require.resolve` is a well-supported method for resolving module paths in CommonJS and ES module contexts.

Reply:
That's not true: https://nodejs.org/docs/latest-v20.x/api/esm.html#importmetaresolvespecifier.

Can you check again if only the change at the `outputPath` variable might be enough?

Reply:
>Can you check again if only the change at the `outputPath` variable might be enough?

Not enough. See the following err

```bash
PS C:\Users\shivams\Desktop\open-source\zudoku> nx run zudoku:build                                                    

> nx run zudoku:"generate:icon-types"

> zudoku@0.23.7 generate:icon-types C:\Users\shivams\Desktop\open-source\zudoku\packages\zudoku
> tsx ./scripts/generate-icon-types.ts
node:fs:442
    return binding.readFileUtf8(path, stringToFlags(options.flag));
                   ^
Error: ENOENT: no such file or directory, open 'C:\C:\Users\shivams\Desktop\open-source\zudoku\node_modules\.pnpm\lucide-react@0.468.0_react@19.0.0\node_modules\lucide-react\dynamicIconImports.js'
```